test/integration/dialects/postgres/query-interface.test.js

```diff
@@ -8,13 +8,162 @@ var chai = require('chai')
+        }).to.throw(/function or trigger used with a parameter without any type/);
+        
+        //requires language
+        expect( function () {
+          self.queryInterface.createFunction('create_job', [{type:'varchar',name:'test'}], 'varchar', null, body, options);
+        }).to.throw(/createFunction missing some parameters. Did you pass functionName, returnType, language and body/);
+
+        //requires body
+        expect( function () {
+          self.queryInterface.createFunction('create_job', [{type:'varchar',name:'test'}], 'varchar', 'plpgsql', null, options);
+        }).to.throw(/createFunction missing some parameters. Did you pass functionName, returnType, language and body/);
+
+      });
+    });
+
+    describe('dropFunction',function () {
+      beforeEach( function () {
+
+        var body = 'return test;';
+        var options = {};
+
+        //make sure we have a droptest function in place.
+        return this.queryInterface.createFunction('droptest', [{type:'varchar',name:'test'}], 'varchar', 'plpgsql', body, options)
+        //suppress errors.. this could fail if the function is already there.. thats ok. 
+        .catch( function (err) { });
+      });
+
+      it('can drop a function', function () {
+        var self = this;
+         
+        //call dropFunction
+        return self.queryInterface.dropFunction('droptest',[{type:'varchar',name:'test'}])
+        //validate
+        .then( function () {
+          return self.sequelize.query('select droptest(\'test\');', { type: self.sequelize.QueryTypes.SELECT })
+          .then( function () {
+            throw new Error('function failed to drop');
+          })
+          .catch( function (err) {
+            //we are expecting to get an error that droptest is nolonger there.
+            var notThere = new RegExp('function droptest.* does not exist');
+            if (notThere.test(err.message)) return;
+            else throw err;
+          });
+        });
+      });
+
+      it('produces an error when missing expected parameters', function () {
+        var self = this;
+        
+        expect( function () {
+          self.queryInterface.dropFunction();
+        }).to.throw(/requires functionName/);
+        
+        expect( function () {
+          self.queryInterface.dropFunction('droptest');
+        }).to.throw(/function parameters array required/);
+
+        expect( function () {
+          self.queryInterface.dropFunction('droptest', [{name:'test'}]);
+        }).to.throw(/function or trigger used with a parameter without any type/);
+
+      });
+    });
+
+    describe('renameFunction', function () {
+
+      it('can rename a function');
+      it('requires parameters array');
+      it('requires old function name');
+      it('requires new function name');
+      it('throws an error if called with no parameter type');
+
+    });
```

Comment:
These are test stubs for tests I felt we should have. My apologies, the last project I was in we put in stubs. I'll remove them.